### ISSUE TITLE
Popover: Screen reader doesn't announce expanded and collapsed states.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2379]
Hi Magma Team,
In the Skills home page, the screen reader does not announce any state change (e.g., expanded or collapsed) after activating the Learn, Roles, Teams control. observation for the header menu related to the PopoverTrigger component
Magma Version - "react-magma-dom": "~4.12.0-next.5",
URL - https://dev.infosec-skills.com/portal/skills/home
Defect - https://infoseci.atlassian.net/browse/AAGL-2082
To solve this problem, you need to fix the following:
Adding aria-haspopup="dialog"directly to the Button ensures the focusable element itself carries both:

aria-expanded={isOpen} 
aria-haspopup="dialog" 

Description
User Experience
Screen reader users rely on state information (e.g., expanded/collapsed) to understand the current status of interactive controls. When the Roles control is activated and its state is not announced, users may not know whether the pop-up has opened or closed, leading to confusion and difficulty in interaction.
Repro Steps
Open the page: https://dev.infosec-skills.com/portal/skills/home and log in.

Navigate to the Roles control using a screen reader.
Activate the control (e.g., press Enter or Space).
Observe the screen reader announcement.
Actual Result
The screen reader does not announce any state change (e.g., expanded or collapsed) after activating the Roles control.
Expected Result
The Roles control should announce its state (e.g., “expanded” when opened and “collapsed” when closed) to ensure users are aware of the control’s current status.
Attachment:
Please refer to the attachment sections.

https://github.com/user-attachments/assets/4c369cd7-03fb-412e-8546-ff55a9382956

https://cengage.slack.com/archives/C8Y4FR63U/p1774501804164049

**NOTE: Inform the team to verify the fix when it's ready.**

**Comment by nikitaorliak-cengage** [comment]
Should be fixed by #2371 -> 
V5: #2380 + #2390  
V4: #2389

**Comment by olehnoskov-cengage** [comment]
**Tested on:**
**- V4:**
Docs: https://next--upbeat-sinoussi-f675aa.netlify.app/api/
Storybook: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/popover--programmatically-opening

Everything works as expected.
**However, VO (macOS + Safari) doesn't announce `expanded/collapsed` states**

**- V5:** 
Docs: https://rc--upbeat-sinoussi-f675aa.netlify.app/api
Storybook: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/popover--programmatically-opening

Everything works as expected.
**However, VO (macOS + Safari) doesn't announce `expanded/collapsed` states**

**Comment by olehnoskov-cengage** [comment]
**Tested on:**
**- V4:**
Docs: https://next--upbeat-sinoussi-f675aa.netlify.app/api/
Storybook: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/popover--programmatically-opening

**- V5:** 
Docs: https://rc--upbeat-sinoussi-f675aa.netlify.app/api
Storybook: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/popover--programmatically-opening

Everything works as expected.

<img width="1478" height="954" alt="Image" src="https://github.com/user-attachments/assets/56f7bca4-3629-4b86-bb78-6f56e40479b7" />
